### PR TITLE
AlternateProtocolsCache: Add Mocks

### PR DESCRIPTION
**Pull request by RyanTheOptimist**
Add mocks of the various AlternateProtocolsCache classes.

Risk Level: Low
Testing: Test-only code
Docs Changes: N/A
Release Notes: N/A
Platform Specific Features: N/A

### CODE REVIEW

File: test/mocks/http/alternate_protocols_cache.cc [hunk 1]
```diff
@@ -0,0 +1,13 @@
+#include "test/mocks/http/alternate_protocols_cache.h"
+
+namespace Envoy {
+namespace Http {
+
+MockAlternateProtocolsCache::~MockAlternateProtocolsCache() = default;
```
Comment (DavidSchinazi) [inline test/mocks/http/alternate_protocols_cache.cc:6]: At this point I'd just define all this in the header file, but up to you. I don't know what the rest of Envoy does in this scenario.

**Review by DavidSchinazi — APPROVED** [review]

File: test/mocks/http/alternate_protocols_cache.cc (hunk 1 above)
Comment (RyanTheOptimist) [reply test/mocks/http/alternate_protocols_cache.cc:6]: Yeah, I share your intuition. I followed the pattern for the other files in test/mock/http and put the = default in the .cc file.

**Review by alyssawilk — APPROVED** [review]
Generally I'd suggest landing this where they're used, but all LGTM and I'll review it either way =P

**Pull request merged**